torch/optim/optimizer.py

```diff
@@ -1049,6 +1084,10 @@ def add_param_group(self, param_group: Dict[str, Any]) -> None:
         param_set: Set[torch.Tensor] = set()
         for group in self.param_groups:
             param_set.update(set(group["params"]))
+            if ('param_names' in param_group) != ('param_names' in group):
+                current_group_txt = 'with names' if 'param_names' in param_group else 'without names'
+                raise ValueError("all optimizer param groups should be with/without names. "
```

Comment:
yea i agree with you this might be an unnecessary constraint but if someone wants this feature later it shouldn't be awful to add it. thanks for the error here!